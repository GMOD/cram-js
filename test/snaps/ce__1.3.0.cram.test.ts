import { testFile } from '../testUtil'
testFile('ce#1.3.0.cram')
