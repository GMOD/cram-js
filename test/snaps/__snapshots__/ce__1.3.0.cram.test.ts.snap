// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`can dump the whole ce#1.3.0.cram without error 1`] = `
[
  {
    "fileId": "-",
    "magic": "CRAM",
    "majorVersion": 3,
    "minorVersion": 0,
  },
  {
    "containerHeader": {
      "_endPosition": 45,
      "_size": 19,
      "alignmentSpan": 0,
      "crc32": 1632985953,
      "landmarks": [
        0,
        133,
      ],
      "length": 208,
      "numBases": 0,
      "numBlocks": 2,
      "numLandmarks": 2,
      "numRecords": 0,
      "recordCounter": 0,
      "refSeqId": 0,
      "refSeqStart": 0,
    },
    "data": [
      {
        "_endPosition": 45,
        "_size": 19,
        "alignmentSpan": 0,
        "crc32": 1632985953,
        "landmarks": [
          0,
          133,
        ],
        "length": 208,
        "numBases": 0,
        "numBlocks": 2,
        "numLandmarks": 2,
        "numRecords": 0,
        "recordCounter": 0,
        "refSeqId": 0,
        "refSeqStart": 0,
      },
      {
        "samHeader": [
          {
            "data": [
              {
                "tag": "SN",
                "value": "CHROMOSOME_I",
              },
              {
                "tag": "LN",
                "value": "1009800",
              },
              {
                "tag": "M5",
                "value": "8ede36131e0dbf3417807e48f77f3ebd",
              },
              {
                "tag": "UR",
                "value": "/home/vadim/bin/htslib/test/ce.fa",
              },
            ],
            "tag": "SQ",
          },
        ],
      },
      {
        "samHeader": [
          {
            "data": [
              {
                "tag": "SN",
                "value": "CHROMOSOME_I",
              },
              {
                "tag": "LN",
                "value": "1009800",
              },
              {
                "tag": "M5",
                "value": "8ede36131e0dbf3417807e48f77f3ebd",
              },
              {
                "tag": "UR",
                "value": "/home/vadim/bin/htslib/test/ce.fa",
              },
            ],
            "tag": "SQ",
          },
        ],
      },
    ],
  },
  {
    "compressionScheme": {
      "APdelta": true,
      "dataSeriesEncoding": {
        "AP": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              0,
            ],
          },
          "parametersBytes": 4,
        },
        "BB": {
          "codecId": 4,
          "parameters": {
            "lengthsEncoding": {
              "codecId": 1,
              "parameters": {
                "blockContentId": 42,
              },
              "parametersBytes": 1,
            },
            "valuesEncoding": {
              "codecId": 1,
              "parameters": {
                "blockContentId": 37,
              },
              "parametersBytes": 1,
            },
          },
          "parametersBytes": 6,
        },
        "BF": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              16,
            ],
          },
          "parametersBytes": 4,
        },
        "BS": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              1,
            ],
          },
          "parametersBytes": 4,
        },
        "CF": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              3,
            ],
          },
          "parametersBytes": 4,
        },
        "DL": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              1,
            ],
          },
          "parametersBytes": 4,
        },
        "FC": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              1,
              1,
            ],
            "numCodes": 2,
            "numLengths": 2,
            "symbols": [
              68,
              88,
            ],
          },
          "parametersBytes": 6,
        },
        "FN": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              6,
            ],
          },
          "parametersBytes": 4,
        },
        "FP": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              1,
              2,
              3,
              3,
            ],
            "numCodes": 4,
            "numLengths": 4,
            "symbols": [
              6,
              5,
              1,
              4,
            ],
          },
          "parametersBytes": 10,
        },
        "IN": {
          "codecId": 5,
          "parameters": {
            "blockContentId": 13,
            "stopByte": 0,
          },
          "parametersBytes": 2,
        },
        "MF": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              0,
            ],
          },
          "parametersBytes": 4,
        },
        "MQ": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              1,
            ],
          },
          "parametersBytes": 4,
        },
        "NP": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              0,
            ],
          },
          "parametersBytes": 4,
        },
        "NS": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              -1,
            ],
          },
          "parametersBytes": 8,
        },
        "QS": {
          "codecId": 1,
          "parameters": {
            "blockContentId": 12,
          },
          "parametersBytes": 1,
        },
        "RG": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              -1,
            ],
          },
          "parametersBytes": 8,
        },
        "RI": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              0,
            ],
          },
          "parametersBytes": 4,
        },
        "RL": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              100,
            ],
          },
          "parametersBytes": 4,
        },
        "RN": {
          "codecId": 5,
          "parameters": {
            "blockContentId": 11,
            "stopByte": 0,
          },
          "parametersBytes": 2,
        },
        "SC": {
          "codecId": 5,
          "parameters": {
            "blockContentId": 14,
            "stopByte": 0,
          },
          "parametersBytes": 2,
        },
        "TL": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              0,
            ],
          },
          "parametersBytes": 4,
        },
        "TS": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              0,
            ],
          },
          "parametersBytes": 4,
        },
      },
      "readNamesIncluded": true,
      "referenceRequired": false,
      "substitutionMatrix": [
        [
          "C",
          "G",
          "T",
          "N",
        ],
        [
          "A",
          "G",
          "T",
          "N",
        ],
        [
          "A",
          "C",
          "T",
          "N",
        ],
        [
          "A",
          "C",
          "G",
          "N",
        ],
        [
          "A",
          "C",
          "G",
          "T",
        ],
      ],
      "tagEncoding": {
        "ASc": {
          "codecId": 4,
          "parameters": {
            "lengthsEncoding": {
              "codecId": 3,
              "parameters": {
                "bitLengths": [
                  0,
                ],
                "numCodes": 1,
                "numLengths": 1,
                "symbols": [
                  1,
                ],
              },
              "parametersBytes": 4,
            },
            "valuesEncoding": {
              "codecId": 1,
              "parameters": {
                "blockContentId": 1,
              },
              "parametersBytes": 1,
            },
          },
          "parametersBytes": 9,
        },
        "XGC": {
          "codecId": 4,
          "parameters": {
            "lengthsEncoding": {
              "codecId": 3,
              "parameters": {
                "bitLengths": [
                  0,
                ],
                "numCodes": 1,
                "numLengths": 1,
                "symbols": [
                  1,
                ],
              },
              "parametersBytes": 4,
            },
            "valuesEncoding": {
              "codecId": 1,
              "parameters": {
                "blockContentId": 1,
              },
              "parametersBytes": 1,
            },
          },
          "parametersBytes": 9,
        },
        "XMC": {
          "codecId": 4,
          "parameters": {
            "lengthsEncoding": {
              "codecId": 3,
              "parameters": {
                "bitLengths": [
                  0,
                ],
                "numCodes": 1,
                "numLengths": 1,
                "symbols": [
                  1,
                ],
              },
              "parametersBytes": 4,
            },
            "valuesEncoding": {
              "codecId": 1,
              "parameters": {
                "blockContentId": 1,
              },
              "parametersBytes": 1,
            },
          },
          "parametersBytes": 9,
        },
        "XNC": {
          "codecId": 4,
          "parameters": {
            "lengthsEncoding": {
              "codecId": 3,
              "parameters": {
                "bitLengths": [
                  0,
                ],
                "numCodes": 1,
                "numLengths": 1,
                "symbols": [
                  1,
                ],
              },
              "parametersBytes": 4,
            },
            "valuesEncoding": {
              "codecId": 1,
              "parameters": {
                "blockContentId": 1,
              },
              "parametersBytes": 1,
            },
          },
          "parametersBytes": 9,
        },
        "XOC": {
          "codecId": 4,
          "parameters": {
            "lengthsEncoding": {
              "codecId": 3,
              "parameters": {
                "bitLengths": [
                  0,
                ],
                "numCodes": 1,
                "numLengths": 1,
                "symbols": [
                  1,
                ],
              },
              "parametersBytes": 4,
            },
            "valuesEncoding": {
              "codecId": 1,
              "parameters": {
                "blockContentId": 1,
              },
              "parametersBytes": 1,
            },
          },
          "parametersBytes": 9,
        },
        "XSc": {
          "codecId": 4,
          "parameters": {
            "lengthsEncoding": {
              "codecId": 3,
              "parameters": {
                "bitLengths": [
                  0,
                ],
                "numCodes": 1,
                "numLengths": 1,
                "symbols": [
                  1,
                ],
              },
              "parametersBytes": 4,
            },
            "valuesEncoding": {
              "codecId": 1,
              "parameters": {
                "blockContentId": 1,
              },
              "parametersBytes": 1,
            },
          },
          "parametersBytes": 9,
        },
        "YTZ": {
          "codecId": 5,
          "parameters": {
            "blockContentId": 9,
            "stopByte": 9,
          },
          "parametersBytes": 2,
        },
      },
      "tagIdsDictionary": [
        [
          "XGC",
          "XMC",
          "XNC",
          "XOC",
          "ASc",
          "XSc",
          "YTZ",
        ],
      ],
    },
    "containerHeader": {
      "_endPosition": 271,
      "_size": 18,
      "alignmentSpan": 101,
      "crc32": 314459724,
      "landmarks": [
        339,
      ],
      "length": 586,
      "numBases": 100,
      "numBlocks": 7,
      "numLandmarks": 1,
      "numRecords": 1,
      "recordCounter": 0,
      "refSeqId": 0,
      "refSeqStart": 2,
    },
    "data": [
      {
        "_endPosition": 271,
        "_size": 18,
        "alignmentSpan": 101,
        "crc32": 314459724,
        "landmarks": [
          339,
        ],
        "length": 586,
        "numBases": 100,
        "numBlocks": 7,
        "numLandmarks": 1,
        "numRecords": 1,
        "recordCounter": 0,
        "refSeqId": 0,
        "refSeqStart": 2,
      },
      {
        "features": [
          {
            "alignmentStart": 2,
            "cramFlags": 3,
            "flags": 16,
            "lengthOnRef": 101,
            "mappingQuality": 1,
            "qualityScores": [
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              31,
              33,
              30,
              23,
              33,
              30,
              33,
              32,
              31,
              31,
              35,
              35,
              33,
              34,
              35,
              35,
              34,
              33,
              34,
              31,
              34,
              35,
              34,
              35,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
            ],
            "readBases": undefined,
            "readFeatures": [
              {
                "code": "X",
                "data": 1,
                "pos": 5,
                "refPos": 6,
              },
              {
                "code": "X",
                "data": 1,
                "pos": 6,
                "refPos": 7,
              },
              {
                "code": "X",
                "data": 1,
                "pos": 12,
                "refPos": 13,
              },
              {
                "code": "X",
                "data": 1,
                "pos": 18,
                "refPos": 19,
              },
              {
                "code": "X",
                "data": 1,
                "pos": 24,
                "refPos": 25,
              },
              {
                "code": "D",
                "data": 1,
                "pos": 28,
                "refPos": 29,
              },
            ],
            "readGroupId": -1,
            "readLength": 100,
            "readName": "SRR065390.14978392",
            "sequenceId": 0,
            "tags": {
              "AS": -18,
              "XG": 1,
              "XM": 5,
              "XN": 0,
              "XO": 1,
              "XS": -18,
              "YT": "UU",
            },
            "templateSize": 0,
            "uniqueId": 616,
          },
        ],
        "header": {
          "_endPosition": 651,
          "_size": 36,
          "compressedSize": 32,
          "compressionMethod": "raw",
          "content": Uint8Array [
            0,
            2,
            101,
            1,
            0,
            5,
            4,
            1,
            9,
            11,
            12,
            255,
            255,
            255,
            255,
            15,
            13,
            53,
            108,
            132,
            13,
            170,
            12,
            148,
            54,
            102,
            168,
            225,
            81,
            162,
            145,
            130,
          ],
          "contentId": 0,
          "contentPosition": 615,
          "contentType": "MAPPED_SLICE_HEADER",
          "crc32": 1039276087,
          "parsedContent": {
            "_endPosition": 303,
            "_size": 32,
            "contentIds": [
              1,
              9,
              11,
              12,
            ],
            "md5": [
              13,
              53,
              108,
              132,
              13,
              170,
              12,
              148,
              54,
              102,
              168,
              225,
              81,
              162,
              145,
              130,
            ],
            "numBlocks": 5,
            "numContentIds": 4,
            "numRecords": 1,
            "recordCounter": 0,
            "refBaseBlockId": -1,
            "refSeqId": 0,
            "refSeqSpan": 101,
            "refSeqStart": 2,
          },
          "uncompressedSize": 32,
        },
      },
    ],
  },
]
`;
